src/Voron/Impl/Scratch/ScratchBufferPool.cs

```diff
@@ -183,7 +183,14 @@ public PageFromScratchBuffer Allocate(LowLevelTransaction tx, int numberOfPages)
 
                 bool createNextFile = false;
 
-                if (current.File.HasDiscontinuousSpaceFor(tx, size, _scratchBuffers.Count))
+                if (tx.IsLazyTransaction)
```

Comment:
this is the easiest solution that works (even if not super efficient in terms of reusing scratch, it keeps us supporting any size and rate of bulk insertions using lazy commits and keeping scratch size on ~500MB).